agent/command/git.go

```diff
@@ -112,6 +116,8 @@ func validateCloneMethod(method string) error {
 
 func (opts cloneOpts) validate() error {
 	catcher := grip.NewBasicCatcher()
+	catcher.NewWhen(opts.useScalar && opts.cloneDepth > 0, "cannot use scalar with clone depth")
```

Comment:
Could you put some comments or make the error message about why we can't? I know we haven't done this before for validation (look at s3.put for reference) but that's why I'm asking. I feel like the context gets lost and I end up looking at this validation and ask 'why can't it be used together' or if a user asks, I have to do a bit of research

Reply:
that's fair, I added a comment with an explanation and a link to the scalar docs

Reply:
On second thought I remember now why I didn't add shallow_clone. It's because [shallow_clone sets the clone depth to 100 ](https://github.com/malikchaya2/evergreen/blob/main/agent/command/git.go#L433)so at this point there's only depth.